pkg/providers/vsphere/vsphere.go

```diff
@@ -199,6 +200,32 @@ func (p *vsphereProvider) UpdateKubeConfig(_ *[]byte, _ string) error {
 	return nil
 }
 
+func (p *vsphereProvider) machineConfigsSpecChanged(ctx context.Context, cc *v1alpha1.Cluster, cluster *types.Cluster, newClusterSpec *cluster.Spec) (bool, error) {
+	machineConfigMap := make(map[string]*v1alpha1.VSphereMachineConfig)
+	for _, config := range p.MachineConfigs(nil) {
```

Comment:
We need to clean this up, either remove the parameter from the function or actually use it instead of referencing the values from the struct.

Reply:
The parameter is currently being used in the [snow provider](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/snow/snow.go#L209-L215), but nowhere else I believe. Is this the model all the providers should adopt? It may be worth creating an issue about it

Reply:
Since this is not an exported method, we have control on what parameters to pass in based on the provider right?